### PR TITLE
[Issue 822] Add Alembic check command

### LINKED ISSUE #822: [Task]:  Add a check that verifies you generated Alembic migrations after changing the database schema
### Summary

When we modify our SQLAlchemy models, we then need to generate the database migration files. Alembic has an `alembic check` command that can verify the local DB is up-to-date

### Acceptance criteria

- [ ] `alembic check` command is run somewhere in the CI/CD process

### PR DESCRIPTION
**Pull request by chouinar**
## Summary
Fixes #822

## Changes proposed
Add `alembic check` to the CI/CD which verifies if the DB migrations are up-to-date.

## Context for reviewers
When we modify our SQLAlchemy models, we need to then run `make db-migrate-create <msg>` to generate the Alembic migration files. `alembic check` makes sure you actually did that. Effectively, the check command just tries to create a new migration and sees if anything would be created, and if so errors. See the alembic docs for more info: https://alembic.sqlalchemy.org/en/latest/autogenerate.html#running-alembic-check-to-test-for-new-upgrade-operations

## Additional information
I added a column in a commit without generating a migration, and the CI/CD caught it: https://github.com/HHS/simpler-grants-gov/actions/runs/7144151215/job/19457159149?pr=836

### CODE REVIEW

File: api/Makefile [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 APP_NAME := grants-api
 
+# Colors for output, can be used as:
```
Comment (acouch) [inline api/Makefile:7]: Should these colors be applied to the other checks?

**Review by acouch — COMMENTED** [review]
(praise) Thanks for adding the [column check](https://github.com/HHS/simpler-grants-gov/pull/836/commits/33ccdfcd3b0a63e1ecf0997a9474a8b287f98a2d) 

(chore) Pls add the new command to the docs: https://github.com/HHS/simpler-grants-gov/blob/main/documentation/api/database/database-management.md

(note) I confirmed this works locally by downgrading to the previous migration and checking

File: .github/workflows/ci-api.yml [hunk 2]
```diff
@@ -18,12 +18,18 @@ jobs:
     steps:
       - uses: actions/checkout@v3
 
+      - name: Initialize the docker containers
+        run: make init
```
[the remark this answers is not in the document]
Comment (acouch) [reply .github/workflows/ci-api.yml:22]: (praise) thx for clarifying

File: api/Makefile (hunk 1 above)
Comment (chouinar) [reply api/Makefile:7]: I think most of our other formatter/linter/test utils do a good job of printing out proper error messages automatically in a clear way (eg. mypy might tell you specifically what line and problem there is)

In this case, it wasn't as clear what the issue was as it just ended:
```
make: *** [Makefile:168: db-check-migrations] Error 255
Error: Process completed with exit code 2.
```
which isn't terribly helpful - the lines above that say what migrations it detected, but at first glance you might not know what you're looking at.

If any of the others aren't clear to someone, happy to add messaging to them, but that would be separate

**Review by acouch — APPROVED** [review]

**Pull request merged**